packages/veritone-react-common/src/components/TranslationEngineOutput/TranslationContent/index.js

```diff
@@ -0,0 +1,265 @@
+  };
+
+  renderContents = () => {
+    let {
+      contents,
+
+      onClick,
+      onRerunProcess,
+
+      dataSegmentClassName,
+      errorSegmentClassName,
+      noDataSegmentClassName,
+
+      neglectableTimeMs,
+
+      mediaPlayerTimeMs,
+      mediaPlayerTimeIntervalMs
+    } = this.props;
+
+    let segments = [];
+    let segmentStatus;
+    let segmentStartTime;
+    let segmentStopTime;
+
+    let clearSegmentStatus = () => {
+      segmentStatus = undefined;
+      segmentStartTime = undefined;
+      segmentStopTime = undefined;
+    };
+
+    let createErrorSegment = () => {
+      segments.push({
+        start: segmentStartTime,
+        stop: segmentStopTime,
+        content: (
+          <ErrorSegment
+            startTimeMs={segmentStartTime}
+            stopTimeMs={segmentStopTime}
+            className={classNames(styles.boxSegment, errorSegmentClassName)}
+            onClick={onRerunProcess}
+            key={
+              'translation-error-' + segmentStartTime + '-' + segmentStopTime
+            }
+          />
+        )
+      });
+
+      segmentStartTime = segmentStopTime;
+      segmentStopTime = undefined;
+    };
+
+    let createNoDataSegment = () => {
+      segments.push({
+        start: segmentStartTime,
+        stop: segmentStopTime,
+        content: (
+          <NoDataSegment
+            startTimeMs={segmentStartTime}
+            stopTimeMs={segmentStopTime}
+            className={classNames(styles.boxSegment, noDataSegmentClassName)}
+            key={
+              'translation-nodata-' + segmentStartTime + '-' + segmentStopTime
+            }
+          />
+        )
+      });
+
+      segmentStartTime = segmentStopTime;
+      segmentStopTime = undefined;
+    };
+
+    let createDataSegment = dataSegments => {
+      segments.push({
+        start: segmentStartTime,
+        stop: segmentStopTime,
+        content: dataSegments
+      });
+
+      segmentStartTime = segmentStopTime;
+      segmentStopTime = undefined;
+    };
+
+    contents.forEach((dataChunk, chunkIndex) => {
+      let chunkStartTime = dataChunk.startTimeMs;
+      let chunkStopTime = dataChunk.stopTimeMs;
+
+      if (segmentStartTime === undefined || segmentStartTime < chunkStartTime) {
+        segmentStartTime = chunkStartTime;
+      }
+
+      let series = sortBy(dataChunk.series, 'startTimeMs', 'stopTimeMs');
+      let dataSegmentEntries = [];
+      // ----Start looping through series content----
+      series.forEach(entry => {
+        let entryStartTime = entry.startTimeMs;
+        let entryStopTime = entry.stopTimeMs;
+        let entryStatus = entry.status || (entry.words ? 'success' : 'nodata');
+
+        if (segmentStatus === undefined) {
+          segmentStatus = entryStatus;
+        } else if (segmentStatus !== entryStatus) {
+          // ----Save Previous Segment----
+          if (segmentStatus === 'success') {
+            createDataSegment(dataSegmentEntries.concat([]));
```

Comment:
use the spread operator to create a copy of the array, i.e - `createDataSegment([...dataSegmentEntries])`